### PR TITLE
feat: Add ToggleSwitch components

### PR DESCRIPTION
**Pull request by maranomynet**
Implementation of the [ToggleSwitch](https://www.figma.com/file/pDczqgdlWxgn3YugWZfe1v/UI-Library-%E2%80%93-%F0%9F%96%A5%EF%B8%8F-Desktop?node-id=2682%3A0) UI element

Although the Figma designs don't show any evidence of labels being considered, these components display a visible label by default, to ensure accessibility, but offer an optional `hideLabel` prop.

(Note: The storybook stories are a bit of a shot in the dark, as we don't really know how to run the storybook server locally – yet.)

## What

This implementation provides support for three different markup patterns, applicable to different use-cases: Checkbox. Button, Link

## Why

Instead of providing a single ToggleSwitch component with a super-complex props type signature, we provide three simple components each focusing on a single markup-/usage-pattern.

* `ToggleSwitchLink` for cases where a route change is neccessary part of the state change involved.
* `ToggleSwitchButton` for cases where the toggler should influence other areas/elements on screen (via `aria-controls=""`, etc.)
* `ToggleSwitchCheckbox` for all other cases where the toggling is really just a fancy-looking checkbox (includes support for old-school `name` and `value` attributes.

## Screenshot

image

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
+import { style } from 'treat'
+import { theme } from '@island.is/island-ui/theme'
+const { spacing, color, typography } = theme
+
+export const toggleSwitchLarge = style({})
+// export const right = style({})
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts:6]: Unhelpful comment?

File: libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+export const toggleSwitchWide = style({
+  width: '100%',
+})
+
+export const text = style({
+  selectors: {
+    // [`${checked} > &`]: {
+    //   fontWeight: typography.medium,
+    // }
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts:54]: Unhelpful comment?

File: libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import { theme } from '@island.is/island-ui/theme'
+const { spacing, color, typography } = theme
+
+export const toggleSwitchLarge = style({})
+// export const right = style({})
+export const toggleSwitchChecked = style({})
+export const toggleSwitchDisabled = style({})
+export const toggleSwitchHiddenLabel = style({})
+
+export const checkbox = style({})
```
Comment (barabrian) [inline libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts:11]: is this being used anywhere?

**Review by barabrian — COMMENTED** [review]
Nice, looks good! It would be nice tho to have some visual aids like screenshots or gifs (but it's understandable if there were issues running storybook, and the figma link helped) 😄 

Only some minor comments

Regarding storybook have you tried running this command `yarn start island-ui-storybook` ?

File: libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitchButton.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  /** By default the button is flagged as a simple `aria-pressed="true|fase"` toggle button.
+   *
+   * If `expander={true}` is used the button changes to use `aria-expanded="true|false"` instead.
+   */
+  expander?: boolean
+
+  'aria-controls'?: string
+}
+
+export const ToggleSwitchButton: FC<ToggleSwitchButtonProps> = (props) => {
```
[the remark this answers is not in the document]
Comment (maranomynet) [reply libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitchButton.tsx:19]: Hah! I'm sold. Thanks!

File: libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts (hunk 2 above)
Comment (maranomynet) [reply libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts:54]: The plan was to leave that in as example/jumping-off point for possible/likely future modifications – as the designer types start making suggestions for tweaks ... as the whole label thing is somewhat lacking visual specs as of yet.

File: libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts (hunk 3 above)
Comment (maranomynet) [reply libs/island-ui/core/src/lib/ToggleSwitch/ToggleSwitch.treat.ts:11]: It used to, but not anymore. Good catch.

**Comment by maranomynet** [comment]
I've added Screenshots to the PR description above.

> Regarding storybook have you tried running this command `yarn start island-ui-storybook`?

I tried grepping the docs and somehow missed this. My bad.
Stories now work.

**Review by barabrian — APPROVED** [review]
LGTM

**Pull request merged**